### PR TITLE
Replace popper.js with floating-ui

### LINKED ISSUE #506: Update from popper.js to floating-ui
The maintainers of popper js have moved on to a lower-level library called [floating-ui](https://floating-ui.com/). The library purports to be smaller, faster, and offer more customization options to users.

We should migrate over to this library, using their [migration guide](https://floating-ui.com/docs/migration), to ensure future support and stability of our popovers.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #506 by @zackkrida 

## Description

This PR replaces the older `popper.js` library with its updated version, `floating-ui`. This is a preparatory step for #2840 and is also useful for updating to Nuxt 3.

## Testing Instructions

The popovers should work as they did before.
The differences are:
RTL placement works correctly. You can see in the homepage snapshots that the popover is aligned with the end of the search type button in both LTR and RTL, whereas in the old popovers, the RTL was aligned with the start of the button.
I removed the `stable` setting for scroll gutter. It was causing the blank space in the inline-end side of the search types popover.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VPopover/meta/VPopover.stories.mdx [hunk 1]
```diff
@@ -6,11 +6,24 @@ import {
   Story,
 } from "@storybook/addon-docs"
 
-import { placements as popoverPlacements } from "@popperjs/core"
```
Comment (AetherUnbound) [inline frontend/src/components/VPopover/meta/VPopover.stories.mdx:9]: Why not use the placements from the new library here?

```
import { Placement as popoverPlacements } from "@floating-ui/dom"
```

File: frontend/src/composables/use-floating-ui.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  watch(
+    [
+      floatingElRef,
+      floatingPropsRefs.triggerElement,
+      floatingPropsRefs.visible,
+    ] as const,
+    ([floatingElement, triggerElement, visible], _, onInvalidate) => {
+      if (!triggerElement || !floatingElement) {
+        cleanup.value?.()
```
Comment (AetherUnbound) [inline frontend/src/composables/use-floating-ui.ts:74]: Woah, you can optionally call a function that might be null? 😮 cool!

**Review by AetherUnbound — COMMENTED** [review]
If popovers should function the same as before, why do there appear to be snapshot differences in this PR? 😮

**Comment by fcoveram** [comment]
> Francisco, can you see what the difference between the snapshots is, and whether it matters?

I see some differences, but since we have pending other UI consistency improvements, we can safely merge this one.

File: frontend/src/composables/use-floating-ui.ts (hunk 2 above)
Comment (dhruvkb) [reply frontend/src/composables/use-floating-ui.ts:74]: This is a TIL for me as well! I sometimes miss syntax sugar like this when working with Python.

**Review by dhruvkb — APPROVED** [review]
The 1px shift is weird but it seems insignificant, but these changes to screenshots make me worry about the repo ballooning up in size.

File: frontend/src/components/VPopover/meta/VPopover.stories.mdx (hunk 1 above)
Comment (obulat) [reply frontend/src/components/VPopover/meta/VPopover.stories.mdx:9]: This imports a type, but we need a value, the list of the placement strings. Unfortunately, there's no list in the library that we could import.

File: frontend/src/composables/use-floating-ui.ts (hunk 2 above)
Comment (obulat) [reply frontend/src/composables/use-floating-ui.ts:74]: I think it was the Copilot's suggestion :)

**Comment by obulat** [comment]
> However the frequent, albeit small, changes to screenshots make me worry about the repo ballooning up in size.

I haven't thought about this. Is there a way of purging the old snapshots from the repository, @dhruvkb?

**Review by AetherUnbound — APPROVED** [review]
🚀

**Pull request merged**